### PR TITLE
Fix empty documentHighlight results due to inconsistent path delimiters

### LINKED ISSUE #94: Issue with documentHighlight feature
Hello,

I'm indirectly using this language server through the [Eclipse WildWebDeveloper tool](https://github.com/eclipse/wildwebdeveloper). I encountered an issue with selection highlighting which we think may be caused by the typescript language server itself (more information [here](https://github.com/eclipse/wildwebdeveloper/issues/104)).

Here is the request that is sent to the server:
```
{
   "jsonrpc":"2.0",
   "id":"12",
   "method":"textDocument/hover",
   "params":{
      "textDocument":{
         "uri":"file:///[...]/popup.js"
      },
      "uri":"file:///[...]/popup.js",
      "position":{
         "line":95,
         "character":32
      }
   }
}
```

_popup.js_ is a file checked out locally, but corresponds to the Javascript document located in [this repository](https://github.com/PyvesB/night-video-tuner/blob/master/popup/popup.js). I'm trying to highlight the `storageObj` variable.

However, the response from the server comes back empty:
```
{
   "jsonrpc":"2.0",
   "id":"12",
   "result":{
      "contents":[

      ]
   }
}
```

Is this to be expected?

Thanks in advance for your help.

Cheers,

Pyves

### PR DESCRIPTION
**Pull request by PyvesB**
Closes #94.

I did not provide any additional unit tests as the only realistic setup to reproduce this issue would be when running the test suite on a Windows machine, which is impractical. The change is small and well documented, it's probably not worth it anyway!

### CODE REVIEW

File: server/src/lsp-server.ts [hunk 1]
```diff
@@ -758,7 +758,9 @@ export class LspServer {
         }
         const result: lsp.DocumentHighlight[] = [];
```
Comment (princemaple) [inline server/src/lsp-server.ts]: probably better move the normalization of the `file` up to here, like

```js
const normalizedFilePath = path.normalize(file);
```

**Comment by princemaple** [comment]
@akosyakov Could you please take a look? thanks

File: server/src/lsp-server.ts (hunk 1 above)
Comment (PyvesB) [reply server/src/lsp-server.ts]: Done, thanks.

**Comment by princemaple** [comment]
Is there an option to request a review from the maintainers? @PyvesB

File: server/src/lsp-server.ts [hunk 2]
```diff
@@ -757,8 +757,11 @@ export class LspServer {
             return []
         }
         const result: lsp.DocumentHighlight[] = [];
+        const normalizedFile = path.normalize(file);
```
Comment (akosyakov) [inline server/src/lsp-server.ts]: should not be necessary, file is already normalized via `URI.parse(uri).fsPath`

File: server/src/lsp-server.ts [hunk 3]
```diff
@@ -757,8 +757,11 @@ export class LspServer {
...
         for (const item of response.body) {
-            if (item.file === file) {
+            // tsp returns item.file with POSIX path delimiters, whereas file is platform specific.
+            // Normalizing both ensures that the comparison is consistent.
+            if (path.normalize(item.file) === normalizedFile) {
```
Comment (akosyakov) [inline server/src/lsp-server.ts]: I would try to use something like `uriToPath(pathToUri(itemfile)) === file`

**Comment by PyvesB** [comment]
Thanks for the feedback @akosyakov! How does the alternative approach look?

**Comment by princemaple** [comment]
@akosyakov any chance this one gets released?

**Pull request merged**